testsuite/features/step_definitions/common_steps.rb

```diff
@@ -282,6 +282,26 @@
   ct.distro_create(distro, '/var/autoinstall/SLES15-SP2-x86_64/DVD1/boot/x86_64/loader/linux', '/var/autoinstall/SLES15-SP2-x86_64/DVD1/boot/x86_64/loader/initrd')
 end
 
+Then(/^create profile "([^"]*)" for distro "([^"]*)" as user "([^"]*)" with password "([^"]*)"$/) do |profile, distro, user, pwd|
+  ct = CobblerTest.new
+  ct.login(user, pwd)
```

Comment:
I wonder if would make sense to have a variable thread scope, that instantiate CobblerTest class and log in, only once. And we reuse it. I would say we did something similar in other scenarios. Something worth to take a look at least.

Reply:
This would be something for https://github.com/SUSE/spacewalk/issues/17382, too.

Reply:
Ok, could you mention it there please? Resolving this conversation.

Reply:
This one is NOT the API, it's the cobbler XML-RPC interface

Reply:
I'm sorry to contradict here, but CobblerTest is a Class to handle server API calls in the same way we have in the classes you worked on. It's just that it points to a different endpoint (`/cobbler_api`).

See here:
```
class CobblerTest
  def initialize(server_address = ENV['SERVER'], server_port = 80, server_path = '/cobbler_api')
    @server_address = server_address
    @server_port = server_port
    @server_path = server_path
    @server = XMLRPC::Client.new(server_address, server_path, server_port)
    raise 'No running server at found at ' + server_address unless running?
  end

  def login(user, pass)
    @token = @server.call('login', user, pass)
  rescue
    raise 'login to cobbler failed' + $ERROR_INFO.to_s
  end
```

So, the design and implementation is identical, and we can follow exactly the same strategy of keep the instance and the log in session.

Reply:
Fine to don't include it on this PR, but we should consider it as next refactor.
Otherwise we are keeping open multiple XMLRPC sockets during all the feature, and we spend time log in on every step.